enterprise/tools/rbesxs/rbesxs.go

```diff
@@ -0,0 +1,222 @@
+
+	target     = flag.String("target", "", "Target to build")
+	doSetup    = flag.Bool("setup", false, "Set to true to perform a setup build. Useful if you want to seed or prepare the remote executors in some way.")
+	setupFlags = flag.String("setup_flags", "", "Comma-separated list of extra flags to add to the setup builds.")
+
+	// TODO(iain): add a --repo flag so when local changes aren't needed the
+	// script can clone and setup the test and control repos.
+	controlRepo  = flag.String("control_repo", "", "Filesystem path of the control repository. May be the same as test if no changed are needed.")
+	controlFlags = flag.String("control_flags", "", "Comma-separated list of extra flags to add to the control builds.")
+	testRepo     = flag.String("test_repo", "", "Filesystem path of the test repository. May be the same as control if no changed are needed.")
+	testFlags    = flag.String("test_flags", "", "Comma-separated list of extra flags to add to the test builds.")
+)
+
+func randomString() string {
+	name, err := random.RandomString(20)
+	if err != nil {
+		log.Fatalf("error generating remote instance name: %s", err)
+	}
+	return name
+}
+
+func getBuildbuddyUrlOrDie(stderr string) string {
+	if !strings.Contains(stderr, "Streaming build results to: ") {
+		log.Fatalf("no buildbuddy url found in stderr: %s", stderr)
+		return ""
+	}
+	split := strings.Split(stderr, "Streaming build results to: ")[1]
+	bburl := strings.Split(split, "\n")[0]
+	if bburl == "" {
+		log.Fatalf("buildbuddy url is empty, stderr: %s", stderr)
+	}
+	_, err := url.ParseRequestURI(bburl)
+	if err != nil {
+		log.Fatalf("%s is not a url: %s", bburl, stderr)
+	}
+	return bburl
+}
+
+func newCommand(target, environment, apiKey, remoteInstanceName, tag string, extraFlags []string) *repb.Command {
+	cmd := []string{"bazel", "build", target}
+	envSuffix := "dev"
+	if environment == "prod" {
+		envSuffix = "io"
+	}
+	// Add these first so the buildbuddy flags take precedence.
+	for i := 0; i < len(extraFlags); i++ {
+		if extraFlags[i] != "" {
+			cmd = append(cmd, extraFlags[i])
+		}
+	}
+	cmd = append(cmd, fmt.Sprintf("--remote_executor=remote.buildbuddy.%s", envSuffix))
+	cmd = append(cmd, fmt.Sprintf("--remote_cache=remote.buildbuddy.%s", envSuffix))
+	cmd = append(cmd, fmt.Sprintf("--bes_backend=remote.buildbuddy.%s", envSuffix))
+	cmd = append(cmd, fmt.Sprintf("--bes_results_url=https://app.buildbuddy.%s/invocation/", envSuffix))
+	cmd = append(cmd, fmt.Sprintf("--remote_header=x-buildbuddy-api-key=%s", apiKey))
+	cmd = append(cmd, fmt.Sprintf("--remote_instance_name=%s", remoteInstanceName))
+	if tag != "" {
+		cmd = append(cmd, fmt.Sprintf("--build_metadata=TAGS=%s", tag))
+	}
+	cmd = append(cmd, "--noremote_accept_cached")
+	return &repb.Command{Arguments: cmd}
+}
+
+func bazelClean(ctx context.Context, workDir string) {
+	result := commandutil.Run(ctx, &repb.Command{Arguments: []string{"bazel", "clean"}}, workDir, nil, nil)
```

Comment:
Oh thanks, didn't exactly understand the semantics of `--async` and thought it might return OK in cases where I wouldn't want to proceed here, but looks like that's not the case.